pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+"""
+filter1d - Time domain filtering of 1-D data tables
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, build_arg_string, fmt_docstring, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    E="end",
+    F="filter",
+)
+def filter1d(data, output_type="pandas", outfile=None, **kwargs):
+    r"""
+    Time domain filtering of 1-D data tables.
+
+    Filter a general time domain for multiple column time
+    series data. The user specifies which column is the time (i.e., the
+    independent variable). The fastest operation
+    occurs when the input time series are equally spaced and have no gaps or
+    outliers and the special options are not needed.
+    Read a table and output as a :class:`numpy.ndarray`,
+    :class:`pandas.DataFrame`, or ASCII file.
+
+    Full option list at :gmt-docs:`filter1d.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    filter : str
+        **type**\ *width*\ [**+l**\|\ **u**].
```

Comment:
See https://github.com/GenericMappingTools/gmt/issues/6599
```suggestion
        **type**\ *width*\ [**+h**].
```